Result/Result.swift

```diff
@@ -181,6 +181,26 @@ public func `try`(function: String = __FUNCTION__, file: String = __FILE__, line
 
 // MARK: - Operators
 
+infix operator <^> {
+	associativity left
+	precedence 140
+}
+
+/// fmap
+public func <^> <T, U, Error> (@noescape transform: T -> U, result: Result<T, Error>) -> Result<U, Error> {
+	return result.map(transform)
+}
+
+infix operator <*> {
+	associativity left
+	precedence 140
+}
+
+/// ap
```

Comment:
Could you please document these operators with their use and relationships to other Swift operators?